GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -658,8 +658,10 @@ private int FindRevisionIndex(ObjectId? objectId)
         /// Returns whether the required revision was found and selected.
         /// </summary>
         /// <param name="objectId">Id of the revision to select.</param>
+        /// <param name="toggleSelection">Toggle if the selected state for the revision.</param>
+        /// <param name="showNoRevisionMsg">Show popup if revision is not visible in grid. Should only be set if the ObjectId is in the repo.</param>
         /// <returns><c>true</c> if the required revision was found and selected, otherwise <c>false</c>.</returns>
-        public bool SetSelectedRevision(ObjectId? objectId, bool toggleSelection = false)
+        public bool SetSelectedRevision(ObjectId? objectId, bool toggleSelection = false, bool showNoRevisionMsg = false)
```

Comment:
```suggestion
        public bool SetSelectedRevision(ObjectId? objectId, bool toggleSelection = false, bool showNoRevisionMsg = true)
```
(As you know,) *I* would rather show the message by default and justify the cases where it is suppressed.

Reply:
I choose the defensive choice: to not change the behavior unless you choose to.
The value is set true almost everywhere.
https://github.com/gitextensions/gitextensions/pull/9833#discussion_r792165847

But we can say that new usage of the code must investigate the behavior and change the defaults.